bpf/lib/nodeport.h

```diff
@@ -1548,6 +1606,16 @@ static __always_inline int dsr_reply_icmp4(struct __ctx_buff *ctx,
 	if (ctx_store_bytes(ctx, off + sizeof(ip), &icmp,
 			    sizeof(icmp), 0) < 0)
 		goto drop_err;
+	if (ctx_store_bytes(ctx, off + sizeof(ip) + sizeof(icmp),
+				&inner_ip_hdr, sizeof(inner_ip_hdr), 0) < 0)
+		goto drop_err;
+	if (inner_ip_hdr.protocol == IPPROTO_UDP ||
```

Comment:
As we can never be in a situation with protocol `ICMP` I would suggest a different pattern and `goto drop_err` if  protocol is different from `UDP/TCP`

Reply:
done